pkgs/os-specific/linux/nvidia-x11/settings.nix

```diff
@@ -45,13 +45,23 @@ let
       "OUTPUTDIR=." # src/libXNVCtrl
     ];
 
+    patches = [
+      # Patch the Makefile to also produce a shared library.
+      (if lib.versionOlder nvidia_x11.settingsVersion "400" then ./libxnvctrl-build-shared-3xx.patch
+      else (fetchpatch {
+        url = "https://github.com/NVIDIA/nvidia-settings/pull/105/commits/0ddfd1d4f871846ff79de6630ddf9f09e9acaf39.patch";
```

Comment:
- Iirc the `pull/xxxx/commits/yyyy` URL may get invalidated after force-pushing or even rebase-merging (I may be wrong)
- The `fetchpatch` now has an updated `all` rule, but the 3xx patch doesn't

Reply:
I think you're right.  I thought that would make it robust against PR changes, but I think GitHub does invalidate commit URLs on a force push, at least after a while.  I think I should back out this commit until the PR is merged.